### DISCUSSION TITLE
Trouble using rect

### DISCUSSION THREAD

**Discussion by nanodeath** [cetz-package/cetz#537] [Q&A] [answered]
Hey, I'm pretty much just getting started with cetz, but I'm having some trouble getting the `rect` function to work at all.

Here's what I'm doing:

```typst
#import "@preview/charged-ieee:0.1.0": ieee
#import "@preview/cetz:0.2.2": *

//...

#canvas(length: 0.8cm, {
  plot.plot(size: (8, 6),
    y-label: "Durability",
    x-label: "Time",
    x-tick-step: none,
    y-min: 0,
    {
      plot.add(((0,40), (5,50), (10, 50), (15, 60), (20, 60), (25, 70)))
      plot.annotate({
        rect((5, 0), (10, 100), fill: rgb(50, 50, 200, 50))
        content((5, 0), [Here])
      })
    })
})
```

But I keep getting `Unexpected argument` at the first argument of `rect`. I'm going by the [manual](https://raw.githubusercontent.com/cetz-package/cetz/stable/manual.pdf) under the `rect` function itself (p.18) as well as the `annotate` example (p.56).

What am I doing wrong?

edit: this seems to be working better, but I still get errors:

```typst
#import "@preview/cetz:0.2.2"
// ...
#cetz.canvas(length: 0.8cm, {
  import cetz.plot
  import cetz.draw: *
  
  plot.plot(size: (8, 6),
    y-label: "Durability",
    x-label: "Time",
    x-tick-step: none,
    y-min: 0,
    {
      plot.add(((0,40), (5,50), (10, 50), (15, 60), (20, 60), (25, 70)))
      plot.add-anchor("fail1", (5, 50))
      plot.annotate({
        rect((5, 0), (10, 100), fill: rgb(50, 50, 200, 50))
        content((5, 0), [Here])
      })
    })
    line("plot.fail1", ((), "|-", (0,1.5)), mark: (start: ">"), name: "line")
})
```

`assertion failed: Unknown element 'plot' in elements ()`

If I remove the draw import, the `plot` error goes away, but the `rect` error comes back.

edit2: Ohh, the failure is because "plot" doesn't refer to the "plot" namespace, it refers to an user-defined thing _named_ "plot", which I missed in the example. What a ride. I think it's working now.

**Comment by fenjalien** [comment, marked as answer]
You want to make sure you import the correct functions and modules at the start of the canvas:
```
#cetz.canvas({
  import cetz.draw: * // imports `rect` and `content` and all the other functions to draw with
  import cetz: plot // Imports the plot module
  
  plot.plot(
     ...
  )
})
```

To use `"plot.fail1"` as an anchor you need to give the plot a name using the `name` keyword argument
```
plot.plot(
  name: "plot",
  ...
)
line("plot.fail1", ...)
```

**Reply by nanodeath** [reply]
Yeah, I'm not sure how I got off to the wrong foot on the imports. I think looking at the examples it's not super obvious that all the other non-module imports need to happen inside the canvas body, not in the outer scope. Learning typst at the same time, if it's not obvious 🙂

**Reply by nanodeath** [reply]
Here's the final result btw 🙌 

![image](https://github.com/cetz-package/cetz/assets/10625/8b1fccf1-6a43-4eca-a34c-b8a656b19823)
